protocols/raft/src/main/java/io/atomix/protocols/raft/storage/snapshot/impl/DefaultSnapshotStore.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2015-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+package io.atomix.protocols.raft.storage.snapshot.impl;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.atomix.protocols.raft.storage.RaftStorage;
+import io.atomix.protocols.raft.storage.snapshot.Snapshot;
+import io.atomix.protocols.raft.storage.snapshot.SnapshotStore;
+import io.atomix.storage.StorageLevel;
+import io.atomix.storage.buffer.FileBuffer;
+import io.atomix.storage.buffer.HeapBuffer;
+import io.atomix.utils.time.WallClockTimestamp;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultSnapshotStore implements SnapshotStore {
```

Comment:
I assume this was just moved and is not new code right?

Reply:
Yep, 99% of this PR is just extracting interfaces.

Reply:
ok :ok_hand: